qcelemental/models/molecule.py

```diff
@@ -723,3 +709,235 @@ def nelectrons(self, ifr: int = None) -> int:
+            geometry computed.
+            `data['mill']` is a AlignmentMill namedtuple with fields
+            (shift, rotation, atommap, mirror) that prescribe the transformation
+            from `concern_mol` and the optimally aligned geometry.
+
+        """
+        from ..molparse.align import B787
+
+        rgeom = np.array(ref_mol.geometry)
+        runiq = np.asarray([
+            hashlib.sha1((sym + str(mas)).encode('utf-8')).hexdigest()
+            for sym, mas in zip(ref_mol.symbols, ref_mol.masses)
+        ])
+        cgeom = np.array(concern_mol.geometry)
+        cmass = np.array(concern_mol.masses)
+        celem = np.array(concern_mol.symbols)
+        celez = np.array(concern_mol.atomic_numbers)
+        cuniq = np.asarray([
+            hashlib.sha1((sym + str(mas)).encode('utf-8')).hexdigest()
+            for sym, mas in zip(concern_mol.symbols, concern_mol.masses)
+        ])
+
+        rmsd, solution = B787(
+            cgeom=cgeom,
+            rgeom=rgeom,
+            cuniq=cuniq,
+            runiq=runiq,
+            do_plot=do_plot,
+            verbose=verbose,
+            atoms_map=atoms_map,
+            run_resorting=run_resorting,
+            mols_align=mols_align,
+            run_to_completion=run_to_completion,
+            run_mirror=run_mirror,
+            uno_cutoff=uno_cutoff)
+
+        ageom, amass, aelem, aelez, auniq = solution.align_system(cgeom, cmass, celem, celez, cuniq, reverse=False)
+        adict = from_arrays(
+            geom=ageom,
+            mass=amass,
+            elem=aelem,
+            elez=aelez,
+            units='Bohr',
+            molecular_charge=concern_mol.molecular_charge,
+            molecular_multiplicity=concern_mol.molecular_multiplicity,
+            fix_com=True,
+            fix_orientation=True)
+        amol = Molecule(validate=False, **to_schema(adict, dtype=2))
+
+        # TODO -- can probably do more with fragments in amol now that
+        #         Mol is something with non-contig frags. frags now discarded.
+
+        assert compare_values(
+            concern_mol.nuclear_repulsion_energy(),
+            amol.nuclear_repulsion_energy(),
+            'Q: concern_mol-->returned_mol NRE uncorrupted',
+            atol=1.e-4,
+            quiet=(verbose > 1))
+        if mols_align:
+            assert compare_values(
+                ref_mol.nuclear_repulsion_energy(),
+                amol.nuclear_repulsion_energy(),
+                'Q: concern_mol-->returned_mol NRE matches ref_mol',
+                atol=1.e-4,
+                quiet=(verbose > 1))
+            assert compare(
+                True,
+                np.allclose(ref_mol.geometry, amol.geometry, atol=4),
+                'Q: concern_mol-->returned_mol geometry matches ref_mol',
+                quiet=(verbose > 1))
+
+        return amol, {'rmsd': rmsd, 'mill': solution}
+
+    def scramble(ref_mol,
```

Comment:
This is a pure test function and should go in tests. If it just scrambled then it would likely be fine.

Reply:
How about a return bool so it could scramble or scramble-and-test? problem is that knowing what to test is best kept internal, but compared to the pure scramble fn in molparse/align, it's nice to have it hooked up to Mol.

Reply:
It doesn't serve any useful purpose to a user as implemented?

Reply:
correct. if the user wanted an instance scrambled, my proposed `def scramble(ref_mol, do_shift, ..., do_test=False) -> Molecule` would return a new instance, scrambled accordingly, or a `bool` if `do_test=True` (`return_bool=True`?)

Reply:
But why have the test here in the first place? This could be a function in the test file rather than here. I don't see other instances of code which have tests within functions.